test/callback/test_diagnostic_hdf5_output.py

```diff
@@ -0,0 +1,154 @@
+"""
+Tests diagnostic callbacks and hdf file output.
+"""
+from thetis import *
+from thetis.callback import VolumeConservation3DCallback
+import h5py
+import shutil
+
+
+def test_callbacks():
+
+    lx = 45000.0
+    ly = 3000.0
+    nx = 25
+    ny = 2
+    mesh2d = RectangleMesh(nx, ny, lx, ly)
+    depth = 50.0
+    elev_amp = 1.0
+    n_layers = 6
+    # estimate of max advective velocity used to estimate time step
+    u_mag = Constant(0.5)
+
+    outputdir = 'outputs'
```

Comment:
You can use the built in pytest fixture `tmpdir` to get a temporary directory that is unique per test and is automatically cleaned up by pytest for output.